agent/php_api.c

```diff
@@ -1578,3 +1578,70 @@ PHP_FUNCTION(newrelic_get_trace_metadata) {
     nr_free(span_id);
   }
 }
+
+/*
+ * Purpose      (New Relic API) (Error Fingerprinting)
+ *              Allows a user to register a PHP callback to be invoked by the
+ *              Agent when the application encounters an error for the purpose
+ *              of returning a string for the 'error.group.name' Agent
+ *              Attribute.
+ *
+ */
+#ifdef TAGS
+void zif_newrelic_set_error_group_callback(void); /* ctags landing pad only */
+void newrelic_set_error_group_callback(void);     /* ctags landing pad only */
+#endif
+PHP_FUNCTION(newrelic_set_error_group_callback) {
+  zend_fcall_info fci;
+  zend_fcall_info_cache fcc;
+
+  NR_UNUSED_RETURN_VALUE_PTR;
+  NR_UNUSED_THIS_PTR;
+  NR_UNUSED_RETURN_VALUE_USED;
+
+  nr_php_api_add_supportability_metric("set_error_group_callback" TSRMLS_CC);
+
+  // Verify that only one argument has been provided to the API (the callback)
+  if (1 != ZEND_NUM_ARGS()) {
+    nrl_warning(NRL_API,
+                "newrelic_set_error_group_callback failure: invalid number of "
+                "parameters");
+    RETURN_FALSE;
+  }
+
+  // Verify that the argument passed to the API is a function
+  // Populate function data
+  if (FAILURE == zend_parse_parameters(ZEND_NUM_ARGS(), "f", &fci, &fcc)) {
+    nrl_warning(
+        NRL_API,
+        "newrelic_set_error_group_callback failure: invalid argument passed");
+    RETURN_FALSE;
+  }
+
+  // Verify the user callback accepts 2 arguments
+  if (2 != fcc.function_handler->common.num_args) {
```

Comment:
Is function_handler guaranteed to be non-null?

Reply:
Yes, otherwise the [zpp call above] (https://github.com/newrelic/newrelic-php-agent/pull/719/files#diff-aaa6dfffdf27a5da3f0c89176d8aa9b62226676fae89c5c2c8e0e1a35db06b66R1614)would have failed and the function returned

Reply:
Added in 1d3ea0d3da9ac486dc728a1ebd6dc9a6518f9d6f